src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js

```diff
@@ -278,6 +288,7 @@ define([
 
         removeHandler: function(id) {
             var data = this.getData();
+            delete this.items[id];
```

Comment:
would set to undefined `this.items[id] = undefined;` delete has a bad performance (see https://jsperf.com/delete-vs-undefined-vs-null/3).

Reply:
Setting it to undefined will not delete the key and if you will each over the items the removedIds will still be there. (https://jsfiddle.net/u1mqLu59/1/)